Update branding to 6.0.0 Alpha1

dougbu opened the pull request:
- hold the TFM back at `net5.0`
- update site extensions to split 5.0 and 6.0
- remove unnecessary `@(SuppressBaselineReference)` items
  - either 3.1.7 baselines we're using don't include reference or still using package
  - fix some comments and `Condition` attributes to make remainder easy to find
  - I'll undo any changes that were over-zealous

dougbu committed:
Update branding to 6.0.0 Alpha1
- hold the TFM back at `net5.0`
- correct `TargetFrameworkVersion` in FrameworkList.xml; don't use `$(AspNetCoreMajorMinorVersion)`
- move `$(DefaultNetCoreTargetFramework)` to eng/Versions.props
  - avoid repeating value in AfterSolutionBuild.targets and Directory.Build.props files
- add new versions to `TemplatePackageInstaller`
- update project template test scripts

dougbu commented on eng/Versions.props:
```diff
@@ -233,12 +237,15 @@
-    <!-- Packages from 2.1/2.2 branches used for site extension build -->
+    <!-- Packages from 2.1, 2.2, 3.1 and 5.0 branches used for site extension build. -->
     <MicrosoftAspNetCoreAzureAppServicesSiteExtension21PackageVersion>2.1.1</MicrosoftAspNetCoreAzureAppServicesSiteExtension21PackageVersion>
     <MicrosoftAspNetCoreAzureAppServicesSiteExtension22PackageVersion>2.2.0</MicrosoftAspNetCoreAzureAppServicesSiteExtension22PackageVersion>
```
Should this be 2.2.5 now? Or, can this package be removed here, in Dependencies.props and in Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj❔

BrennanConroy replied:
I'm all for removing anything 2.2
Lets ping some folks to make sure they're ok with us removing 2.2 from the site extension
@Pilchie @shirhatti @bradygaster

dougbu commented on eng/Versions.props:
```diff
@@ -233,12 +237,15 @@
+    <!-- Packages from 2.1, 2.2, 3.1 and 5.0 branches used for site extension build. -->
     <MicrosoftAspNetCoreAzureAppServicesSiteExtension21PackageVersion>2.1.1</MicrosoftAspNetCoreAzureAppServicesSiteExtension21PackageVersion>
     <MicrosoftAspNetCoreAzureAppServicesSiteExtension22PackageVersion>2.2.0</MicrosoftAspNetCoreAzureAppServicesSiteExtension22PackageVersion>
     <MicrosoftAspNetCoreAzureAppServicesSiteExtension31PackageVersion>3.1.3-servicing-20163-14</MicrosoftAspNetCoreAzureAppServicesSiteExtension31PackageVersion>
```
Should this be 3.1.7-servicing-20372-13 now❔ That shipped (sort of) with 3.1.7.

dougbu replied:
@BrennanConroy is this your call❔ If not you, who❔

BrennanConroy replied:
Likely my call, I think it doesn't matter if it stays or updates (assuming the 3.1.3-servicing bits stay in the feed). I'll double check tomorrow.

BrennanConroy replied:
We probably should keep this as the latest. Anyway to automate that?

dougbu replied:
Doubt this is possible to automate given oddities with circular subscriptions. But, let's chat w/ @mmitche and @wtgodbe in the 3.1.9 timeframe. Could you file an issue while I'm thinking of other problems with this PR please❔ (Placing it in https://github.com/orgs/dotnet/projects/10 will help us remember.)

dougbu commented on eng/Versions.props:
```diff
@@ -233,12 +237,15 @@
     <MicrosoftAspNetCoreAzureAppServicesSiteExtension31PackageVersion>3.1.3-servicing-20163-14</MicrosoftAspNetCoreAzureAppServicesSiteExtension31PackageVersion>
     <MicrosoftAspNetCoreAzureAppServicesSiteExtension31x64PackageVersion>$(MicrosoftAspNetCoreAzureAppServicesSiteExtension31PackageVersion)</MicrosoftAspNetCoreAzureAppServicesSiteExtension31x64PackageVersion>
     <MicrosoftAspNetCoreAzureAppServicesSiteExtension31x86PackageVersion>$(MicrosoftAspNetCoreAzureAppServicesSiteExtension31PackageVersion)</MicrosoftAspNetCoreAzureAppServicesSiteExtension31x86PackageVersion>
+    <MicrosoftAspNetCoreAzureAppServicesSiteExtension50PackageVersion>5.0.0-preview-7-20365-19</MicrosoftAspNetCoreAzureAppServicesSiteExtension50PackageVersion>
```
/fyi this is the version that shipped (sort of) with 5.0.0-preview-7.

dougbu commented on src/SiteExtensions/LoggingAggregate/src/Microsoft.AspNetCore.AzureAppServices.SiteExtension/Microsoft.AspNetCore.AzureAppServices.SiteExtension.csproj:
```diff
@@ -53,10 +53,14 @@
+        When updating AspNetCoreMajorVersion or AspNetCoreMinorVersion, add the following SiteExtension packages to
+        the list above and bump the version here.
+      -->
+      <PackageReference Include="Microsoft.AspNetCore.AzureAppServices.SiteExtension.6.0.x86"
```
This and the `@(PackageReference)` items in release/5.0 need to use `5.0`. Is that on your team @BrennanConroy❔

BrennanConroy replied:
Did you mean to put this comment on the other side of the diff?

I'll take a look at 5.0 and update 3.1 -> 5.0 if needed

dougbu replied:
Much appreciated❕ Hopefully 3.1 isn't busted in this respect too.

BrennanConroy replied:
I don't think it is, that's the one that was fixed a couple months ago and I think the merge forward kept this as 3.1

BrennanConroy replied:
Technically 5.0 should probably be updated, but it's currently producing the correct package. I guess I'll update it for consistency.

dougbu commented:
/azp run

dougbu commented:
/fyi This PR needs more work to get the tests passing ☹️

dougbu committed:
Update / add 6.0 package versions in test expectations
- basically, remove test assumptions that branding == target TFM
- `TestData` will need another update once 6.0.0.0 assembly versions arrive from dotnet/runtime

dougbu commented on eng/AfterSolutionBuild.targets:
```diff
@@ -15,7 +15,7 @@
       Files="@(_BuildOutput)"
       FrameworkOnlyPackages="@(AspNetCoreAppReference)"
-      SharedFrameworkTargetFramework="net$(AspNetCoreMajorVersion).$(AspNetCoreMinorVersion)" />
+      SharedFrameworkTargetFramework="net5.0" />
```
Use of `$(AspNetCoreMajorVersion)` or `$(AspNetCoreMajorMinorVersion)` was only occasionally correct. Since the concepts aren't related, we shouldn't change this back in the future

dougbu commented on eng/targets/Helix.props:
```diff
@@ -22,7 +22,7 @@
     <LoggingTestingDisableFileLogging Condition="'$(IsHelixJob)' == 'true'">false</LoggingTestingDisableFileLogging>
     <NodeVersion>10.15.3</NodeVersion>
-    <AppRuntimeVersion>5.0.0-ci</AppRuntimeVersion>
+    <AppRuntimeVersion>6.0.0-ci</AppRuntimeVersion>
```
Note to self: If we need another update, use `$(AspNetCoreMajorMinorVersion).$(AspNetCorePatchVersion)` here

dougbu replied:
Property removed instead

dougbu commented on src/Identity/Extensions.Stores/src/Microsoft.Extensions.Identity.Stores.csproj:
```diff
@@ -13,7 +13,6 @@
     <Reference Include="Microsoft.Extensions.Identity.Core" />
-    <SuppressBaselineReference Include="System.ComponentModel.Annotations" />
-  </ItemGroup>
+ </ItemGroup>
```
Note to self: If we need another iteration, fix indentation here

dougbu commented:
Internal build https://dev.azure.com/dnceng/internal/_build/results?buildId=778919 should help confirm changes in site extensions build

dougbu commented:
:up::date: though I doubt tests will work without further changes. I'm building locally to see if I can get more information about template failures in previous builds. Suspect we have additional places where TFMs get mixed up with package versions.

Worst case, we might need a runtime update containing dotnet/runtime#40950. I triggered that (daily) subscription just in case.

dougbu commented:
@pranavkm I seem to remember you looking at test failures like and that you resolved them. 
``` text
The active test run was aborted. Reason: Test host process crashed : Unhandled exception. System.InvalidOperationException: There is no currently active test.
```
Any idea why I'm seeing them in the gRPC interop tests e.g. https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-24983-merge-c76e36077044499e8d/InteropTests--net5.0/console.74b37675.log?sv=2019-02-02&se=2020-09-09T20%3A09%3A24Z&sr=c&sp=rl&sig=89ThpfLWPzA57TSNhBUDUXyo5Zl7VM%2FE5dV4aEzYBKI%3D❔

dougbu commented:
@HaoK I'm seeing gRPC test failures on other platforms that seem to indicate the newly-built 6.0.0-ci copy of Microsoft.AspNetCore.App.Runtime was installed to the wrong location. What needs to change❔

For example, from https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-24983-merge-e4e47cb7c8b8484ca9/InteropTests--net5.0/console.ab71631b.log?sv=2019-02-02&se=2020-09-09T20%3A08%3A14Z&sr=c&sp=rl&sig=YqU17c32Tp8rRlY3EZdxU9Vo%2BSHqmJYs5K7gpC6QeuE%3D:
``` text
Found AspNetRuntime: Microsoft.AspNetCore.App.Runtime.win-x64.6.0.0-ci.nupkg, extracting *.txt,json,dll to /datadisks/disk1/work/B8880A11/w/B18309D8/e/.dotnet28088/shared/Microsoft.AspNetCore.App/5.0.0-rc.1.20417.14
```

HaoK commented:
The versions haven't been updated yet? `-sdk 5.0.100-rc.1.20379.10 --runtime 5.0.0-rc.1.20417.14`

https://github.com/dotnet/aspnetcore/blob/master/eng/targets/Helix.targets#L120  I think we use MicrosoftNETCoreAppRuntimeVersion

dougbu commented:
> The versions haven't been updated yet? `-sdk 5.0.100-rc.1.20379.10 --runtime 5.0.0-rc.1.20417.14`

We should be able to rebrand independent of the dotnet/runtime repo. Helix.targets and the scripts used on the Helix agents should _not_ use `$(MicrosoftNETCoreAppRuntimeVersion)` when installing Microsoft.AspNetCore.App.Runtime. Guess I've got another thing to separate in this PR&hellip;

----

/fyi the same messed up install location for the new Microsoft.AspNetCore.App.Runtime bits is hurting the project template tests too e.g. https://helixre107v0xdeko0k025g8.blob.core.windows.net/dotnet-aspnetcore-refs-pull-24983-merge-56d6b0427e38425e94/ProjectTemplates.Tests--net5.0/console.2800c965.log?sv=2019-02-02&se=2020-09-09T20%3A09%3A23Z&sr=c&sp=rl&sig=fJ7Q1BIJ1qJNSQxW8J2SP801r31rmo3kHfbakOY0szw%3D

On Windows, the project template downstream errors are usually "failed to run EF migrations" due to an inability to load the Microsoft.AspNetCore.Diagnostics.Abstractions.dll assembly that Microsoft.AspNetCore.Diagnostics.EntityFrameworkCore.dll needs. The gRPC downstream errors are thrown `TimeoutException`s.

HaoK commented:
Right both GRPC and templates depend on the aspnet runtime, that currently use that version, i think if you just replace that with whatever the aspnet specific version is, things shoudl work

dougbu commented:
@HaoK please have a look @ the last commit

dougbu commented on eng/targets/Helix.targets:
```diff
@@ -16,10 +16,6 @@
-  </ItemGroup>
-
   <ItemGroup Condition="'$(IsHelixJob)' == 'true' AND '$(TestDependsOnAspNetRuntime)' == 'true'">
     <HelixContent Include="$(RepoRoot)artifacts\packages\Release\Shipping\*-ci.nupkg" />
```
What needs every package we ship❔ And, does including these packages in the correlation payload mean they're available when restoring packages❔ This isn't reliable in any case because it's often evaluated before projects build or publish.

HaoK replied:
The template tests use a wide variety of packages we build, rather than listing them all, it seems reasonable for us to just install the ones we build, these aren't in the correlation payload, they show up in the workitem root

dougbu replied:
Ah,
``` text
nuget add source {Options.HELIX_WORKITEM_ROOT} --configfile NuGet.config 
```

dougbu commented:
> @HaoK please have a look @ the last commit

Now the second-to-last commit

dougbu commented:
New test internal build is https://dev.azure.com/dnceng/internal/_build/results?buildId=781173 Previous one didn't uncover any issues in the packages but I'd like to double-check

HaoK commented:
If you say SharedFxVersion is the right version to use, I'm guessing it is :)  Looks fine to me

dougbu commented:
Thanx @HaoK❕ 

@JunTaoLuo or @wtgodbe please review this PR before we get too far into tomorrow. And, code owners please chime in if something in here skews your world.

dougbu commented:
`dotnet ef` commands in project template tests are failing with similar `FileLoadException`s to the Blazor WASM tests. Project template exceptions are about Microsoft.AspNetCore.Diagnostics.Abstractions.dll. Unfortunately these problems did not result in `*.dmp` files.

gRPC interop tests now failing with `BadImageFormatException`s. `*.dmp` files may help here.

cartermp commented on src/ProjectTemplates/Web.ProjectTemplates/Microsoft.DotNet.Web.ProjectTemplates.csproj:
```diff
@@ -58,6 +58,7 @@
     <GeneratedContent Include="WebApi-CSharp.csproj.in" OutputPath="content/WebApi-CSharp/Company.WebApplication1.csproj" />
     <GeneratedContent Include="WebApi-FSharp.fsproj.in" OutputPath="content/WebApi-FSharp/Company.WebApplication1.fsproj" />
     <GeneratedContent Include="Worker-CSharp.csproj.in" OutputPath="content/Worker-CSharp/Company.Application1.csproj" />
+    <GeneratedContent Include="Worker-FSharp.csproj.in" OutputPath="content/Worker-FSharp/Company.Application1.csproj" />
```
Actually two problems:

1. This not being in the template means that the worker service does not build
2. It should be a .fsproj, like in the PR here: https://github.com/dotnet/aspnetcore/pull/22042/files#diff-77ac0b0b902172169ed9b7307e56cf52 - so something went awry

dougbu replied:
Given problems getting this PR working, I'm thinking about separating this one-line fix into another PR. For the first point, the `WorkerTemplateAsync()` test checks only C#. I'll try adding another test if I create a new PR tonight or (more likely) tomorrow. (Want to see if I can find the cause of the image loading issues w/ Fusion logging first.)

I fixed the second problem.

dougbu replied:
Moved this change to #25180.

dougbu commented:
FYI @captainsafia @javiercn @pranavkm I'm having no luck getting Fusion logs for the Blazor WASM `FileLoadException` and haven't downloaded a `*.dmp` from the most recent failures on Helix.

dougbu commented:
:up::date: to remove the three commits I moved to #25180 and #25181. None of those commits is _required_ in this PR. They were here only because I found the problems while working on the branding and the branding / TFM confusion.

dougbu commented:
/azp run

dougbu committed:
Update `Condition`s on `@(SuppressBaselineReference)` items to 6.0

dougbu commented:
/fyi @mmitche this probably needs another iteration but is much closer than it was.

dougbu commented on src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:
```diff
@@ -45,5 +52,30 @@
+  <Target Name="_UpdateSharedFrameworkAssemblies"
+      AfterTargets="ResolveFrameworkReferences"
+      BeforeTargets="ResolveTargetingPackAssets;_PrepareForReadyToRunCompilation;ResolveFrameworkReferencesDesignTime;_WarnAboutRedundantRef"
+      DependsOnTargets="ResolveFrameworkReferences">
```
@rainersigwald I added `AfterTargets` because the `ResolveFrameworkReferences` has a `Returns` and thought that indicated it might get used from an outer build. But, that doesn't seem to be the case, at least in the cases I checked. In addition, this project has just a single target framework as seen above -- https://github.com/dotnet/aspnetcore/pull/24983/files#diff-eef634c0ab4b1d011d44f35afc0ed324R4
So, should we need both `AfterTargets` and `DependsOnTargets`❔

FYI the overall aim here is to make sure we fix the resolved targeting pack and framework reference metadata before anything tries to use it.

rainersigwald replied:
Having it listed in both is not unreasonable.

`AfterTargets` means "whenever `ResolveFrameworkReferences` runs, run this right afterward."

`DependsOnTargets` means "if this is going to run, run `ResolveFrameworkReferences` first."

In the normal flow of things, `ResolveFrameworkReferences` runs because of other dependencies (in the common targets/the SDK). So your `AfterTargets` probably generally hooks to that, and might be sufficient on its own.

`DependsOnTargets` would make sense if the target dependency flow goes through this target--because one of its `BeforeTargets` wants to run, so this runs, and the `DependsOn` forces `ResolveFrameworkReferences` to run.

dougbu commented on src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:
```diff
@@ -13,8 +13,9 @@
+      _UpdateSharedFrameworkAssemblies.target.
     -->
     <UseAspNetCoreSharedRuntime>true</UseAspNetCoreSharedRuntime>
     <DoNotApplyWorkaroundsToMicrosoftAspNetCoreApp>true</DoNotApplyWorkaroundsToMicrosoftAspNetCoreApp>
```
@pranavkm this project is a mess and I don't remember why it needs these hacks. the .nuspec will contain just the framework reference no matter what we do (due to the `RemoveSharedFrameworkOnlyRefsFromNuspec` fixes in eng/AfterSolutionBuild.targets) and anything that lands in the .deps.json file (or whatever) will be overridden using the targeting pack's PackageOverrides.txt file in user's projects i.e. packages won't be downloaded.

Please remind me: Why do we need this unique special case❔

dougbu committed:
Undo some of #24816
- allow sharedFx assemblies in Microsoft.AspNetCore.Components.WebAssembly.DevServer package
- see #26448 about reducing this baggage later in 6.0

dougbu committed:
Skip IndividualAuth tests on Helix
- #26776

dougbu commented:
/ping reviewers

This PR addresses additional version confusions that have crept into the repo, specifically mixing the default .NET TFM with the shared framework version. Switching to 6.0.0 does not mean we're targeting net6.0 (yet).

This PR also introduces follow-up items #26448 and #26776.

dougbu commented:
PR is green.

Only aspnetcore-quarantined-pr tests failed in last validation runs. Failures in https://dev.azure.com/dnceng/public/_build/results?buildId=848137 look very similar to those in https://dev.azure.com/dnceng/public/_build/results?buildId=848129 (the last rolling build of the master branch), including the Helix Linux and Windows timeouts. IOW there's no new failures in this PR compared to the existing master branch.

JunTaoLuo commented on src/Framework/Directory.Build.props:
```diff
@@ -7,15 +7,15 @@
     <FrameworkListRootAttributes Include="Name" Value="$(AspNetCoreAppFrameworkBrandName)" />
     <FrameworkListRootAttributes Include="TargetFrameworkIdentifier" Value="$(NETCoreAppFrameworkIdentifier)" />
-    <FrameworkListRootAttributes Include="TargetFrameworkVersion" Value="$(AspNetCoreMajorMinorVersion)" />
+    <FrameworkListRootAttributes Include="TargetFrameworkVersion" Value="$(DefaultNetCoreTargetFramework.Substring(3))" />
```
This doesn't look resilient. Granted we probably won't change the format netX.Y to something else, I guess I'm okay with it.

JunTaoLuo commented on src/Components/WebAssembly/DevServer/src/Microsoft.AspNetCore.Components.WebAssembly.DevServer.csproj:
```diff
@@ -12,24 +12,10 @@
   </PropertyGroup>
 
   <ItemGroup>
-    <FrameworkReference Include="Microsoft.AspNetCore.App" />
```
Why did this reference change? I'm not too sure about the changes in this file but as long as @pranavkm signs off it should be fine.

mmitche replied:
Probably putting an error above that indicating that we expect "netN" so that we catch this location if something changes.

dougbu replied:
The whole project changed because we have a mess of conflicts because this project attempted to use AspNetCore bits from the SDK as well as reference projects elsewhere in the repo. @pranavkm signed off on doing this in Teams.

 #26448 is about trimming the package _correctly_ at some point in the 6.0 train but only if this project doesn't move elsewhere.
